extensions/samples/electron-webpack/src/AuthProvider.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { shell } from "electron";
+import {
+    PublicClientApplication,
+    LogLevel,
+    AccountInfo,
+    SilentFlowRequest,
+    AuthenticationResult,
+    InteractiveRequest,
+} from "@azure/msal-node";
+import {
+    DataProtectionScope,
+    Environment,
+    PersistenceCreator,
+    PersistenceCachePlugin,
+} from "@azure/msal-node-extensions";
+import path from "path";
+
+export default class AuthProvider {
+    private clientApplication: PublicClientApplication;
+    private account: AccountInfo;
+
+    constructor() {}
+
+    static async build(authConfig: any): Promise<any> {
+        try {
+            const authProvider = new AuthProvider();
+            authProvider.account = null;
+            const cachePath = path.join(
+                Environment.getUserRootDirectory(),
+                "./cache.json"
+            );
+
+            const persistenceConfiguration = {
+                cachePath,
+                dataProtectionScope: DataProtectionScope.CurrentUser,
+                serviceName: "test-msal-electron-service",
+                accountName: "test-msal-electron-account",
+                usePlaintextFileOnLinux: false,
+            };
+
+            const persistence = await PersistenceCreator.createPersistence(
+                persistenceConfiguration
+            );
+            authProvider.clientApplication = new PublicClientApplication({
+                auth: authConfig.auth,
+                // This hooks up the cross-platform cache into MSAL
+                cache: {
+                    cachePlugin: new PersistenceCachePlugin(persistence),
+                },
+                system: {
+                    loggerOptions: {
+                        loggerCallback(loglevel, message, containsPii) {
```

Comment:
nit: since this is a TS file, can you add types here (and everywhere else you're missing types)?

example: loglevel: LogLevel, message: string, containsPii: Boolean